cmd/sonobuoy/app/gen_test.go

```diff
@@ -276,9 +279,104 @@ func TestResolveConfig(t *testing.T) {
 				t.Errorf("expected PluginSelections %v but got %v", tc.expected.PluginSelections, conf.PluginSelections)
 			}
+
 			if !reflect.DeepEqual(conf.Resources, tc.expected.Resources) {
 				t.Errorf("expected resources %v but got %v", tc.expected.Resources, conf.Resources)
 			}
 		})
 	}
 }
+
+func TestResolveConfigSonobuoyNamespace(t *testing.T) {
+	defaultPluginSearchPath := config.New().PluginSearchPath
+	defaultAggr := plugin.AggregationConfig{TimeoutSeconds: 10800}
+	defaultSonobuoyNamespace := new(bool)
+	*defaultSonobuoyNamespace = true
+
+	tcs := []struct {
+		name     string
+		input    *genFlags
+		expected *config.Config
+		cliInput string
+	}{
+		{
+			name:     "nil boolean config will be overwritten by default value",
+			input:    &genFlags{
+				sonobuoyConfig: SonobuoyConfig{
+					Config: config.Config{
+						Limits: config.LimitConfig{
+							PodLogs: config.PodLogLimits{
+								SonobuoyNamespace: nil,
+							},
+						},
+					},
+					raw: RawInput,
+				},
+
+			},
+			expected: &config.Config{
+				Namespace:        "heptio-sonobuoy",
+				WorkerImage:      "gcr.io/heptio-images/sonobuoy:" + buildinfo.Version,
+				ImagePullPolicy:  "IfNotPresent",
+				PluginSearchPath: defaultPluginSearchPath,
+				Aggregation:      defaultAggr,
+				PluginSelections: nil,
+				Resources:        config.DefaultResources,
+				Limits: config.LimitConfig{
+					PodLogs: config.PodLogLimits{
+						SonobuoyNamespace: defaultSonobuoyNamespace,
+					},
+				},
+			},
+		},
+		{
+			name:     "non-nil boolean config should be preserved",
+			input:    &genFlags{
+				sonobuoyConfig: SonobuoyConfig{
+					Config: config.Config{
+						Limits: config.LimitConfig{
+							PodLogs: config.PodLogLimits{
+								SonobuoyNamespace: &[]bool{false}[0],
+							},
+						},
+					},
+					raw: RawInput,
+				},
+			},
+			expected: &config.Config{
+				Namespace:        "heptio-sonobuoy",
+				WorkerImage:      "gcr.io/heptio-images/sonobuoy:" + buildinfo.Version,
+				ImagePullPolicy:  "IfNotPresent",
+				PluginSearchPath: defaultPluginSearchPath,
+				Aggregation:      defaultAggr,
+				PluginSelections: nil,
+				Resources:        config.DefaultResources,
+				Limits: config.LimitConfig{
+					PodLogs: config.PodLogLimits{
+						SonobuoyNamespace: &[]bool{false}[0],
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			// Simulate parsing of input via CLI. Making this optional to avoid complicating
+			// setup for other tests which just explicitly set the values.
+			if len(tc.cliInput) > 0 {
+				fs := GenFlagSet(tc.input, EnabledRBACMode)
+				if err := fs.Parse(strings.Split(tc.cliInput, " ")); err != nil {
+					t.Fatalf("Failed to parse CLI input %q: %v", tc.cliInput, err)
+				}
+			}
+
+			conf := tc.input.resolveConfig()
+
+			if *conf.Limits.PodLogs.SonobuoyNamespace != *tc.expected.Limits.PodLogs.SonobuoyNamespace {
```

Comment:
nit: You're defining the test as if it is going to test any/all fields of the config and also test config vs CLI interactions.

If here you're just testing the one section of the config you can get rid of the CLI input field/parsing and remove lots of the structure on the resulting object (no need to define `expected` to be the entire struct if you just want to confirm the `*bool` value.

Likewise, you could make it less verbose by not defining a whole genconfig object for eadch test if you just want to test the effect of changing the `SonobuoyNamespace` or `PodLogLimits`